File: app/src/main/java/com/govnokoder/velotracker/ui/main/PageStart.java
package com.govnokoder.velotracker.ui.main;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import androidx.fragment.app.Fragment;

import com.govnokoder.velotracker.BL.Controller.TrainingController;
import com.govnokoder.velotracker.BL.Model.Training;
import com.govnokoder.velotracker.R;

import java.util.List;

public class PageStart extends Fragment {

    private int pageNumber;
    private Button StartButton;
    private LinearLayout linearStat, linearLastTraining;
    private TextView totalDistanceText, lastTrainingDateText, lastTrainingTimeText, lastTrainingDistanceText, lastTrainingAverageSpeedText;



    public interface onSomeEventListener {
        public void startTraining();
        public void openLastTraining();
        public void openStatistic();
    }

    onSomeEventListener someEventListener;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try {
            someEventListener = (onSomeEventListener) context;
        } catch (ClassCastException e) {
            throw new ClassCastException(context.toString() + " must implement onSomeEventListener");
        }
    }



    public static PageStart newInstance(int page) {
        PageStart fragment = new PageStart();
        Bundle args=new Bundle();
        args.putInt("num", page);
        fragment.setArguments(args);
        return fragment;
    }

    public PageStart() { }

    @Override
    public void onResume() {
        super.onResume();
        TrainingController trainingController = new TrainingController(getContext());
        List<Training> trainings = trainingController.LoadTrainingsData();
        if(trainings != null && trainings.size() > 0){
            Training lastTraining = trainings.get(trainings.size() - 1);
            double totalDist = 0;
            for (Training training : trainings) {
                totalDist += training.Distance;
            }
            totalDistanceText.setText(Double.toString(Training.round(totalDist, 2)) + getString(R.string.km));
            lastTrainingTimeText.setText(lastTraining.Time.toString());
            lastTrainingAverageSpeedText.setText(Double.toString(Training.round(lastTraining.AverageSpeed, 1)) + getString(R.string.kph));
            lastTrainingDistanceText.setText(Double.toString(Training.round(lastTraining.Distance, 2)) + getString(R.string.km));
            lastTrainingDateText.setText(lastTraining.Date.toString());
        }
        else {
            lastTrainingTimeText.setText("00:00");
            lastTrainingAverageSpeedText.setText("0 " + getString(R.string.kph));
            lastTrainingDistanceText.setText("0 " + getString(R.string.km));
            lastTrainingDateText.setText("00/00/0000");
            totalDistanceText.setText("0 " + getString(R.string.km));
        }
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        pageNumber = getArguments() != null ? getArguments().getInt("num") : 1;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View result = inflater.inflate(R.layout.start_page, container, false);
        linearLastTraining = result.findViewById(R.id.linearLastTraining);
        linearLastTraining.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                someEventListener.openLastTraining();
            }
        });
        StartButton = result.findViewById(R.id.startButton);
        StartButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                    someEventListener.startTraining();
            }
        });

        linearStat = result.findViewById(R.id.linearStat);
        linearStat.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                someEventListener.openStatistic();
            }
        });

        totalDistanceText = result.findViewById(R.id.totalDistanceTextV);
        lastTrainingDateText = result.findViewById(R.id.lastTrainingDateTextV);
        lastTrainingDistanceText = result.findViewById(R.id.lastDistanceTextV);
        lastTrainingAverageSpeedText = result.findViewById(R.id.lastAverageSpeedTextV);
        lastTrainingTimeText = result.findViewById(R.id.lastTimeTextV);
        return result;
    }
}

File: app/src/main/java/com/coursework/velotracker/ui/main/PageStart.kt
package com.coursework.velotracker.ui.main

import android.annotation.SuppressLint
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.coursework.velotracker.AppConstants
import com.coursework.velotracker.BL.Controller.TrainingController
import com.coursework.velotracker.BL.Model.Extensions.round
import com.coursework.velotracker.BL.Model.Extensions.toStringExtension
import com.coursework.velotracker.BL.Model.Training.TrainingStatistics
import com.coursework.velotracker.R
import com.coursework.velotracker.databinding.HistoryPageBinding
import com.coursework.velotracker.databinding.StartPageBinding


class PageStart(): Fragment() {

    private var _binding: StartPageBinding? = null
    private val binding get() = _binding!!
    private var pageNumber = 1
    private lateinit var onSomeEventListener: OnSomeEventListener

    interface OnSomeEventListener{
        fun startTraining()
        fun openLastTraining()
        fun openStatistic()
    }

    companion object{
         fun newInstance(page: Int): PageStart{
            val fragment = PageStart()
            val args = Bundle()
            args.putInt("num", page)
            fragment.arguments = args
            return fragment
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        pageNumber = arguments?.getInt("num")?:1
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        _binding = StartPageBinding.inflate(inflater, container, false)
        initFunc()
        return binding.root
    }

    private fun initFunc(){
        binding.linearLastTraining.setOnClickListener { onSomeEventListener.openLastTraining() }
        binding.startButton.setOnClickListener(View.OnClickListener { onSomeEventListener.startTraining() })
        binding.linearStat.setOnClickListener(View.OnClickListener { onSomeEventListener.openStatistic() })
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        try{
            onSomeEventListener = context as OnSomeEventListener
        }catch (e: java.lang.ClassCastException){
            throw ClassCastException("$context must implement OnSomeEventListener")
        }
    }

    override fun onResume() {
        super.onResume()
        val trainings = loadTrainingStatistics()
        if(trainings?.size!! > 0){
            setTrainingValues(trainings)
        }else{
            setDefaultTrainingValues()
        }
    }

    private fun loadTrainingStatistics(): MutableList<TrainingStatistics>? {
        val trainingController = context?.let { TrainingController(it) }
        return trainingController?.loadTrainings()
    }

    @SuppressLint("SetTextI18n")
    fun setTrainingValues(trainings: MutableList<TrainingStatistics>){
        val lastTraining = trainings[trainings.size - 1]
        var totalDistance = 0.0
        trainings.forEach{
            totalDistance+= it.totalDistance
        }
        binding.totalDistanceTextV.text = round(totalDistance, 2).toString() + getString(R.string.km)
        binding.lastTimeTextV.text = lastTraining.totalTime.toStringExtension()
        binding.lastTrainingDateTextV.text = lastTraining.date.toStringExtension(AppConstants.DATE_FORMAT)
        binding.lastAverageSpeedTextV.text = round(lastTraining.averageSpeed, 1).toString() + getString(R.string.kph)
        binding.lastDistanceTextV.text = round(lastTraining.totalDistance, 2).toString() + getString(R.string.km)
    }

    @SuppressLint("SetTextI18n")
    fun setDefaultTrainingValues(){
        binding.lastTimeTextV.text = "00:00"
        binding.lastAverageSpeedTextV.text = "0 " + getString(R.string.kph)
        binding.lastDistanceTextV.text = "0 " + getString(R.string.km)
        binding.lastTrainingDateTextV.text = "00/00/00";
        binding.totalDistanceTextV.text = "0 " + getString(R.string.km)
    }
}